src/controllers/prove.ts

```diff
@@ -0,0 +1,40 @@
+import { Body, Controller, Ctx, Get, Params, Post } from 'amala'
+import { Context } from 'koa'
+import { Job, JobModel } from '@/models/Job'
+import { notFound } from '@hapi/boom'
+import Inputs from '@/validators/Inputs'
+import JobStatus from '@/models/JobStatus'
+import ProofResultParams from '@/validators/ProofResultParam'
+
+@Controller('/prove')
+export default class ProveController {
+  @Post('/')
+  async proof(@Body({ required: true }) input: Inputs) {
+    const job = await JobModel.create({ input })
+    job.input = undefined
+    const result: { job: Job; position?: number } = { job }
+    if (job.status === JobStatus.scheduled) {
+      result.position = await JobModel.countDocuments({
+        status: JobStatus.scheduled,
+        createdAt: { $lt: job.createdAt },
+      })
+    }
+    return result
+  }
+  @Get('/:id')
```

Comment:
Maybe we should make some kind of secret key so that users can't fetch other people's proofs?